src/librustc_mir/repr.rs

```diff
@@ -473,33 +468,25 @@ impl<'tcx> Debug for Lvalue<'tcx> {
         use self::Lvalue::*;
 
         match *self {
-            Var(id) =>
-                write!(fmt,"Var({:?})", id),
-            Arg(id) =>
-                write!(fmt,"Arg({:?})", id),
-            Temp(id) =>
-                write!(fmt,"Temp({:?})", id),
-            Static(id) =>
-                write!(fmt,"Static({:?})", id),
-            ReturnPointer =>
-                write!(fmt,"ReturnPointer"),
-            Projection(ref data) =>
-                match data.elem {
-                    ProjectionElem::Downcast(_, variant_index) =>
-                        write!(fmt,"({:?} as {:?})", data.base, variant_index),
-                    ProjectionElem::Deref =>
-                        write!(fmt,"(*{:?})", data.base),
-                    ProjectionElem::Field(Field::Named(name)) =>
-                        write!(fmt,"{:?}.{:?}", data.base, name),
-                    ProjectionElem::Field(Field::Indexed(index)) =>
-                        write!(fmt,"{:?}.{:?}", data.base, index),
-                    ProjectionElem::Index(ref index) =>
-                        write!(fmt,"{:?}[{:?}]", data.base, index),
-                    ProjectionElem::ConstantIndex { offset, min_length, from_end: false } =>
-                        write!(fmt,"{:?}[{:?} of {:?}]", data.base, offset, min_length),
-                    ProjectionElem::ConstantIndex { offset, min_length, from_end: true } =>
-                        write!(fmt,"{:?}[-{:?} of {:?}]", data.base, offset, min_length),
-                },
+            Var(id) => write!(fmt, "Var({:?})", id),
+            Arg(id) => write!(fmt, "Arg({:?})", id),
+            Temp(id) => write!(fmt, "Temp({:?})", id),
+            Static(id) => write!(fmt, "Static({:?})", id),
+            ReturnPointer => write!(fmt, "ReturnPointer"),
+            Projection(ref data) => match data.elem {
+                ProjectionElem::Downcast(_, variant_index) =>
+                    write!(fmt, "({:?} as {:?})", data.base, variant_index),
+                ProjectionElem::Deref => write!(fmt, "(*{:?})", data.base),
```

Comment:
I think there's a case to be mased for consistency in cases like these: i.e. either all match arms go on a new line, or none of them. Probably plenty of place where this heuristic breaks though